flink-runtime/src/test/java/org/apache/flink/runtime/externalresource/ExternalResourceUtilsTest.java

```diff
@@ -0,0 +1,247 @@
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingFailedExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
+
+		final Map<String, ExternalResourceDriver> externalResourceDrivers = ExternalResourceUtils.externalResourceDriversFromConfig(config, testingPluginManagerPluginManager);
+
+		assertThat(externalResourceDrivers.entrySet(), is(empty()));
+	}
+
+	@Test
+	public void testGetExternalResourceInfoProvider() {
+		final Configuration config = new Configuration();
+		final Map<Class<?>, Iterator<?>> plugins = new HashMap<>();
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		final String driverFactoryClassName = TestingExternalResourceDriverFactory.class.getName();
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), RESOURCE_AMOUNT_1);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
+
+		final ExternalResourceInfoProvider externalResourceInfoProvider = ExternalResourceUtils.getExternalResourceInfoProvider(config, testingPluginManagerPluginManager);
+
+		assertNotNull(externalResourceInfoProvider.getExternalResources().get(RESOURCE_NAME_1));
+	}
+
+	@Test
+	public void testGetExternalResourceInfoProviderWithIllegalAmount() {
+		final Configuration config = new Configuration();
+		final Map<Class<?>, Iterator<?>> plugins = new HashMap<>();
+		plugins.put(ExternalResourceDriverFactory.class, IteratorUtils.singletonIterator(new TestingExternalResourceDriverFactory()));
+		final PluginManager testingPluginManagerPluginManager = new TestingPluginManager(plugins);
+
+		final String driverFactoryClassName = TestingExternalResourceDriverFactory.class.getName();
+		config.set(ExternalResourceOptions.EXTERNAL_RESOURCE_LIST, Collections.singletonList(RESOURCE_NAME_1));
+		config.setLong(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX), 0);
+		config.setString(ExternalResourceOptions.keyWithResourceNameAndSuffix(RESOURCE_NAME_1, ExternalResourceOptions.EXTERNAL_RESOURCE_DRIVER_FACTORY_SUFFIX), driverFactoryClassName);
```

Comment:
Same here. Due to the fact that `getExternalResourceInfoProvider` does a lot of things, we need to set up all these things in order to test that we fail if we provide an illegal amount. I would be simpler if could directly provide the drivers and the amount map to a method which creates the `ExternalResourceInfoProvider`.